src/hooks/useCommitsLimit.ts

```diff
@@ -0,0 +1,25 @@
+import { MutableRefObject, useEffect, useRef } from "react";
```

Comment:
Same comment about folder structure. 
Although i would argue this should be colocated in the commits folder since this only applies to that page.